libdeno/snapshot_creator.cc

```diff
@@ -40,5 +40,9 @@ int main(int argc, char** argv) {
   std::ofstream file_(snapshot_out_bin, std::ios::binary);
```

Comment:
I don't understand why we clone the snapshot into a `std::string` before writing to file. This is a waste. (And also, there is a risk: `std::string` assumes  no `\0` in the middle of array. )
Why not remove `snapshot_str` and use `file_.write(snapshot.data_ptr, snapshot.data_len);` directly?

Reply:
> `std::string` assumes no `\0` in the middle of array.

I don't think that's true per se (it's only assumed when initializing the std::string from a c string).

Reply:
@F001 Fixed - thanks.